baybe/searchspace/discrete.py

```diff
@@ -230,6 +230,85 @@ def discrete_parameter_factory(
 
         return cls(parameters=parameters, exp_rep=df, empty_encoding=empty_encoding)
 
+    @classmethod
+    def from_simplex(
+        cls,
+        parameters: List[NumericalDiscreteParameter],
+        total: float,
+        boundary_only: bool = False,
+        tolerance: float = 1e-6,
+    ) -> SubspaceDiscrete:
+        """Efficiently create discrete simplex subspaces.
+
+        The same result can be achieved using
+        :meth:`baybe.searchspace.discrete.from_product` in combination with appropriate
+        sum constraints. However, such an approach is inefficient because the Cartesian
+        product involved creates an exponentially large set of candidates, most of
+        which do not satisfy the simplex constraints and must be subsequently be
+        filtered out by the method.
+
+        By contrast, this method uses a shortcut that removes invalid candidates
+        already during the creation of parameter combinations, resulting in a
+        significantly faster construction.
+
+        Args:
+            parameters: The parameter to be used for the simplex construction.
+            total: The desired sum of the parameter values defining the simplex size.
+            boundary_only: Flag determining whether to keep only parameter
+                configurations on the simplex boundary.
+            tolerance: Numerical tolerance used to validate the simplex constraint.
+
+        Raises:
+            ValueError: If the passed parameters are not suitable for a simplex
+                construction.
+
+        Returns:
+            The created simplex subspace.
+        """
+        # Validate parameter types
+        if not (all(isinstance(p, NumericalDiscreteParameter) for p in parameters)):
+            raise ValueError(
+                f"All parameters passed to '{cls.from_simplex.__name__}' "
+                f"must be of type '{NumericalDiscreteParameter.__name__}'."
+            )
+
+        # Validate non-negativity
+        min_values = [min(p.values) for p in parameters]
+        if not (min(min_values) >= 0.0):
```

Comment:
I guess theoretically one could also extend this to arbitrary values, but I restricted it for now for the following reasons:
* The standard definition of a "simplex" is indeed for non-negative values only!
* The motivation for the method were the mixture use cases we already have, and there you'll always have non-negativity
* Most critically, the shortcut I took for the creation would be much more complicated if you also allowed negative values, because then you can still add parameters and reduce the total sum. 

So mostly for the last reason, this is omitted. I'm open to extend the method in the future, but this would definitely be a second PR and I'd only do it if we have a use case that needs it (otherwise, we have more important issues to solve 😄 )

If you are OK with that, please resolve